include/mysql/service_mysql_keyring.h

```diff
@@ -16,10 +16,72 @@
 #ifndef MYSQL_SERVICE_MYSQL_PLUGIN_KEYRING_INCLUDED
 #define MYSQL_SERVICE_MYSQL_PLUGIN_KEYRING_INCLUDED
 
+#ifndef MYSQL_ABI_CHECK
+#include <stdlib.h>
```

Comment:
It is not very good to limit MYSQL_ABI_CHECK like that, but I also don't see an easy way out since you need size_t, maybe @percona-ysorokin will have an idea?

Reply:
I have moved it to mysys. Mysql sql/ seems like a better place for this function, but I think innodb will later also need access to those definitions.